src/ripple/ledger/CashDiff.h

```diff
@@ -0,0 +1,131 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2016 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_LEDGER_CASHDIFF_H_INCLUDED
+#define RIPPLE_LEDGER_CASHDIFF_H_INCLUDED
+
+#include <ripple/protocol/STAmount.h>
+#include <memory>                       // std::unique_ptr
+
+namespace ripple {
+
+class ReadView;
+
+namespace detail {
+
+class ApplyStateTable;
+
+}
+
+// Used by CashDiff to specify filters applied while processing differences.
+// Entries are bit flags that can be ANDed and ORed.
+enum class CashFilter : std::uint8_t
+{
+    none = 0x0,
+    treatZeroOfferAsDeletion = 0x1
+};
+inline CashFilter operator| (CashFilter lhs, CashFilter rhs)
+{
+    using ul_t = std::underlying_type<CashFilter>::type;
+    return static_cast<CashFilter>(
+        static_cast<ul_t>(lhs) | static_cast<ul_t>(rhs));
+}
+inline CashFilter operator& (CashFilter lhs, CashFilter rhs)
+{
+    using ul_t = std::underlying_type<CashFilter>::type;
+    return static_cast<CashFilter>(
+        static_cast<ul_t>(lhs) & static_cast<ul_t>(rhs));
+}
+
+//------------------------------------------------------------------------------
+
+// A class to identify differences between two ApplyStateTable instances
+// for debugging.
+class CashDiff
```

Comment:
This belongs in the unit test source directory since its only used for testing. And because its not part of any public APIs, the requirements are relaxed when choosing identifier names. Its fine the way it is (but the code has to be moved).